lib/trino-orc/src/main/java/io/trino/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -288,7 +288,7 @@ public void writeBlock(Block block)
             rowGroupValueCount++;
             totalValueCount++;
 
-            if (!block.isNull(position)) {
+            if (!DictionaryBuilder.isNullIndex(index)) {
```

Comment:
@leeyh0216 could you check what the generated assembly looks like for the `io.trino.orc.writer.SliceDictionaryColumnWriter#writeBlock` method? especially if the `io.trino.orc.writer.DictionaryBuilder#putIfAbsent` is inlined?
I suspect that if the `putifAbsent` is inlined, this change will not improve perf. We still may do it to not depend on the jit inlining decision

Reply:
@lukasz-stec , Thank you for your comment.

As in your comment, I confirmed that the code is inlined.  I share the test results and close this pull request.

Test Code: 
```
    @Test
    public void test() {
        int entries = 100000;
        Random random = new Random();
        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(null, entries, 32 * entries);
        for(int i = 0; i < entries; i++) {
            if(random.nextInt(1000) % 17 == 0) {
                blockBuilder.appendNull();
            }
            else {
                Slice strSlice = Slices.utf8Slice(UUID.randomUUID().toString());
                blockBuilder.writeBytes(strSlice, 0, strSlice.length());
            }
            blockBuilder.closeEntry();
        }
        Block block = blockBuilder.build();
        SliceDictionaryColumnWriter writer = new SliceDictionaryColumnWriter(
                ROOT_COLUMN,
                VARCHAR,
                CompressionKind.NONE,
                toIntExact(DEFAULT_MAX_COMPRESSION_BUFFER_SIZE.toBytes()),
                () -> new StringStatisticsBuilder(toIntExact(DEFAULT_MAX_STRING_STATISTICS_LIMIT.toBytes()), new NoOpBloomFilterBuilder()));
        writer.beginRowGroup();
        writer.writeBlock(block);
        writer.finishRowGroup();
    }
```

I ran it with `-XX:+PrintCompilation -XX:+UnlockDiagnosticVMOptions -XX:+PrintInlining` option and it prints like 
```
 3184 1515       3       io.trino.orc.writer.DictionaryBuilder::putIfAbsent (81 bytes)
                              @ 3   java.util.Objects::requireNonNull (15 bytes)   inline
                                @ 9   java.lang.NullPointerException::<init> (6 bytes)   don't inline Throwable constructors
                              @ 9   io.trino.spi.block.AbstractVariableWidthBlock::isNull (11 bytes)   inline
                                @ 2   io.trino.spi.block.BlockUtil::checkReadablePosition (11 bytes)   inline
                                  @ 2   io.trino.spi.block.Block::getPositionCount (0 bytes)   no static binding
                                  @ 7   io.trino.spi.block.BlockUtil::checkValidPosition (41 bytes)   callee is too large
                                @ 7   io.trino.spi.block.AbstractVariableWidthBlock::isEntryNull (0 bytes)   no static binding
```

Some non-inlined code exists.

As described in the PR Description, more assembly codes are generated, but during the performance test, i could not confirm a big difference between before and after modification.

Thank you for your review again